src/main/java/org/assertj/core/internal/Maps.java

```diff
@@ -614,4 +625,44 @@ private void assertNotNull(AssertionInfo info, Map<?, ?> actual) {
     if (entries.length == 0) throw failures.failure(info, shouldBeEmpty(actual));
   }
 
+  private static <T, U> void doCheckOnMapMissingContents(Predicate<T> contentsCondition, T predicateOperand,
+                                                         NonaryFunction onConditionMet) {
+    try {
+      if (contentsCondition.test(predicateOperand)) onConditionMet.apply();
+    } catch (NullPointerException npe) {
+      if (predicateOperand == null) {
+        /*
+         * The specification of Map allows for map implementations that refuse null keys or values. In this
+         * case, certain operations may throw NPEs if passed a null argument (containsKey and
+         * containsValue, for example). If we catch an NPE here and the argument to the predicate
+         * is null, we will assume we are dealing with a predicate that uses such a method and proceed as if
+         * the predicate had yielded true.
+         */
+        onConditionMet.apply();
+      } else {
+        // The NPE was thrown for a different reason than described above. Rethrow it.
+        throw npe;
+      }
+    }
+  }
+  
+  private static <T, U> void doCheckOnMapPresentContents(Predicate<T> contentsCondition, T predicateOperand,
+                                                         NonaryFunction onConditionMet) {
+    try {
+      if (contentsCondition.test(predicateOperand)) onConditionMet.apply();
+    } catch (NullPointerException npe) {
+      /*
+       * The specification of Map allows for map implementations that refuse null keys or values. In this
+       * case, certain operations may throw NPEs if passed a null argument (containsKey and
+       * containsValue, for example). If we catch an NPE here and the argument to the predicate
+       * is null, we will assume we are dealing with a predicate that uses such a method and ignore the 
+       * exception as the map will definitely not contain null.
+       */
+      if (predicateOperand != null) {
+        // The NPE was thrown for a different reason than described above. Rethrow it.
+        throw npe;
+      }
+    }
+  }
```

Comment:
I have the feeling that these two methods are a bit over the threshold of necessary complexity, probably two "smart" wrappers for `containsKey` and `containsValue` could also be enough. I'll experiment in this direction.